### PR TITLE
Fix for the issue https://github.com/Azure/azure-powershell/issues/5192 

### PR DESCRIPTION
**Pull request by darshanhs90**
Fix for issue https://github.com/Azure/azure-powershell/issues/5192

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#making-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Resources/Commands.Resources/Models.Authorization/AuthorizationClientExtensions.cs [hunk 1]
```diff
@@ -117,7 +118,19 @@ private static IEnumerable<PSRoleAssignment> ToPSRoleAssignments(this IEnumerabl
 
             List<string> objectIds = new List<string>();
             objectIds.AddRange(assignments.Select(r => r.PrincipalId.ToString()));
-            List<PSADObject> adObjects = activeDirectoryClient.GetObjectsByObjectId(objectIds);
+            List<PSADObject> adObjects = null;
+            try
+            {
+                adObjects = activeDirectoryClient.GetObjectsByObjectId(objectIds);
+            }
+            catch (CloudException ce) {
```
Comment (cormacpayne) [inline src/ResourceManager/Resources/Commands.Resources/Models.Authorization/AuthorizationClientExtensions.cs]: @darshanhs90 I think we should add a separate function that will check for the below condition:

```cs
public bool IsFooError(CloduException ce)
{
    if (ce.Response != null && ce.Response.StatusCode == HttpStatusCode.Unauthorized &&
        ce.Error != null && ce.Error.Code != null && string.Equals(ce.Error.Code, "Authorization_RequestDenied", StringComparison.OrdinalIgnoreCase))
    {
        return true;
    }
    
    return false
}
```

This allows us to handle this special case and return a custom exception, while allowing the other exceptions to be thrown as they usually would and preserve their error message and content:

```cs
catch (CloudException ex) when (IsFooError(ex))
{
    throw new InvalidOperationException(ProjectResources.InSufficientGraphPermission);
}
```

File: src/ResourceManager/Resources/Commands.Resources/Models.Authorization/AuthorizationClientExtensions.cs [hunk 2]
```diff
@@ -117,7 +118,19 @@ private static IEnumerable<PSRoleAssignment> ToPSRoleAssignments(this IEnumerabl
...
+                if (ce.Response.StatusCode == HttpStatusCode.Unauthorized && ce.Error.Code.Equals("Authorization_RequestDenied",StringComparison.OrdinalIgnoreCase))
```
Comment (cormacpayne) [inline src/ResourceManager/Resources/Commands.Resources/Models.Authorization/AuthorizationClientExtensions.cs]: @darshanhs90 nit: should we make the "Authorization_RequestDenied` string a constant?

Comment (darshanhs90) [reply src/ResourceManager/Resources/Commands.Resources/Models.Authorization/AuthorizationClientExtensions.cs]: Done

File: src/ResourceManager/Resources/Commands.Resources/Models.Authorization/AuthorizationClientExtensions.cs (hunk 1 above)
Comment (darshanhs90) [reply src/ResourceManager/Resources/Commands.Resources/Models.Authorization/AuthorizationClientExtensions.cs]: Done

Comment (cormacpayne) [reply src/ResourceManager/Resources/Commands.Resources/Models.Authorization/AuthorizationClientExtensions.cs]: @darshanhs90 this is almost correct -- we want to make sure that we're only catching this special type of exception, which is why we'd use the `when` check:

```cs
catch (CloudException ex) when (IsAuthorizationDeniedException(ex))
{
    throw new InvalidOperationException(ProjectResources.InsufficientGraphPermission);
}
```

**Comment by cormacpayne** [comment]
On-demand: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell-demand/504/

**Pull request merged**